app-emulation/hercules-sdl: new package, add 4.7.0

matoro opened the pull request:
This is the current active fork of Hercules, maintained by SoftDevLabs.

I have been maintaining this ebuild in my overlay, and running this fork on ppc64 hardware, for ~6 months now.  There have been some cleanups to satisfy pkgcheck, where relevant.

The "submodules" here are static libraries required only at build-time.  However, they build using cmake, while the main code builds using autotools.  I could not identify any clean way of combining the two using cmake.eclass, so they are separate packages here.

Upstream does not tag these modules on release.  This has been brought to their attention in https://github.com/SDL-Hercules-390/hyperion/issues/530, but no progress yet.  Therefore I have included a script, `files/gen_hashes.sh`, which identifies the relevant commit hashes based on the release date of the main tag.

Having this fork available will pave the way for eventually last-riting `app-emulation/hercules`, which will clear up a number of bugs due to its dead upstream:  [bug 521032](https://bugs.gentoo.org/show_bug.cgi?id=521032), [bug 828460](https://bugs.gentoo.org/show_bug.cgi?id=828460), [bug 879629](https://bugs.gentoo.org/show_bug.cgi?id=828460).

floppym reviewed (changes requested):
The author on your commits is `Matoro Mahri <matoro@users.noreply.github.com>`, but the signed-off-by is `Matoro Mahri <matoro_gentoo@matoro.tk>`. Please ensure these match.

matoro commented:
> The author on your commits is `Matoro Mahri <matoro@users.noreply.github.com>`, but the signed-off-by is `Matoro Mahri <matoro_gentoo@matoro.tk>`. Please ensure these match.

Oops, this is so old that some commits were made before I switched over.  Fixed!
